grammar.js

```diff
@@ -1,157 +1,228 @@
+/* eslint-disable-next-line spaced-comment */
+/// <reference types="tree-sitter-cli/dsl" />
+// @ts-check
+
+const primitives = [
+  'blob',
+  'boolean',
+  'byte',
+  'document',
+  'double',
+  'float',
+  'integer',
+  'long',
+  'short',
+  'string',
+  'timestamp',
+  'bigInteger',
+  'bigDecimal',
+];
+
 module.exports = grammar({
   name: 'smithy',
-  extras: $ => [
+  extras: ($) => [
     /\s/,
-    ',',
+    $.comment,
     $.documentation_comment,
-    $.comment
+    ',', // commas are whitespace in smithy
   ],
 
   rules: {
     // TODO: add the actual grammar rules
-    source_file: $ =>
+    source_file: ($) =>
```

Comment:
Why is this `($)` necessary?
I've looked at several other grammars and it's not being used anywhere, so I would suggest reverting it - especially given it makes the diff much larger than it needs to be.

In general, I really like the CI and tests part, but the grammar changes seem to be dramatic and obscured by the `$ => ($)` addition. Could you please revert that so it's clear what you are actually changing?

Here's what I'm talking about: 
<img width="1099" alt="image" src="https://user-images.githubusercontent.com/1052965/215419185-eeb4d221-dde3-43b0-8144-0e1729214d5b.png">

This became part of a larger diff even though the sections are basically the same.

Reply:
It stems from the eslint linter/code formatter I added, that is just the spec they use. I'lll revert it now